src/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobWriter.cs

```diff
@@ -348,18 +348,18 @@ public void WriteDateTime(DateTime value)
         /// <remarks>
         /// References may be small (2B) or large (4B).
         /// </remarks>
-        public void WriteReference(uint reference, int size)
+        public void WriteReference(int reference, bool isSmall)
         {
-            Debug.Assert(size == 2 || size == 4);
+            // This code is a very hot path, hence we don't check if the reference actually fits 2B.
 
-            if (size == 2)
+            if (isSmall)
             {
                 Debug.Assert((ushort)reference == reference);
-                WriteUInt16((ushort)reference);
+                WriteUInt16(unchecked((ushort)reference));
```

Comment:
Why `unchecked` given the above assert? I assume we won't turn on the perf hit of checked arithmetic in release, and to me `unchecked` implies overflow is by design where as overflow here would be a bug in the caller, no?